framework/src/base/MooseApp.C

```diff
@@ -3503,20 +3527,22 @@ MooseApp::determineLibtorchDeviceType(const MooseEnum & device_enum) const
   {
 #ifdef __linux__
     if (!torch::cuda::is_available())
-      mooseError("--libtorch-device=cuda: CUDA is not available");
+      mooseError(
+          "--compute-device=cuda: CUDA support is not available in the linked libtorch library");
     return torch::kCUDA;
 #else
-    mooseError("--libtorch-device=cuda: CUDA is not supported on your platform");
+    mooseError("--compute-device=cuda: CUDA is not supported on your platform");
 #endif
   }
   else if (device_enum == "mps")
   {
 #ifdef __APPLE__
     if (!torch::mps::is_available())
-      mooseError("--libtorch-device=mps: MPS is not available");
+      mooseError(
+          "--compute-device=mps: MPS support is not available in the linked libtorch library");
     return torch::kMPS;
 #else
-    mooseError("--libtorch-device=mps: MPS is not supported on your platform");
+    mooseError("--compute-device=mps: MPS is not supported on your platform");
```

Comment:
since we are populating from a more general device parameter, we'll need to change the assert that follows this code to actually give a meaningful error for all METHODs `else if (device_enum != "cpu")`